applications/nrf5340_audio/src/modules/hw_codec.c

```diff
@@ -316,3 +319,96 @@ int hw_codec_init(void)
 
 	return 0;
 }
+
+static int cmd_input(const struct shell *shell, size_t argc, char **argv)
+{
+	int ret;
+	uint8_t idx;
+
+	enum hw_codec_input {
+		LINE_IN,
+		PDM_MIC,
+		NUM_INPUT,
+	};
+
+	if (argc != 2) {
+		shell_error(shell, "Wrong number of arguments provided %d", argc);
+		return -EINVAL;
+	}
+
+	if (!isdigit((int)argv[1][0])) {
+		shell_error(shell, "Supplied argument is not numeric");
+		return -EINVAL;
+	}
+
+	idx = strtoul(argv[1], NULL, BASE_10);
+
+	if (idx >= NUM_INPUT) {
+		shell_error(shell, "Selected input out of range");
+		return -EINVAL;
+	}
+
+	if ((CONFIG_AUDIO_DEV == GATEWAY) && IS_ENABLED(CONFIG_AUDIO_SOURCE_USB)) {
+		shell_error(shell, "Can't select PDM mic if audio source is USB");
+		return -EINVAL;
+	}
+	if ((CONFIG_AUDIO_DEV == HEADSET) && !IS_ENABLED(CONFIG_STREAM_BIDIRECTIONAL)) {
+		shell_error(shell, "Cant select input if headset is not in bidirectional stream");
+		return -EINVAL;
+	}
+
+	if (idx == LINE_IN) {
+		if (CONFIG_AUDIO_DEV == HEADSET) {
+			ret = cs47l63_comm_reg_conf_write(input_enable, ARRAY_SIZE(input_enable));
+			if (ret) {
+				shell_error(shell, "Failed to enable line in");
```

Comment:
"line-in" instead of "line in" could be better

Reply:
We use "LINE-IN" in cs47l63_reg_conf.h

Reply:
Fixed